src/bitmessagekivy/baseclass/addressbook.py

```diff
@@ -0,0 +1,171 @@
+        else:
+            self.ids.ml.add_widget(self.default_label_when_empty())
+
+    def set_mdList(self, start_index, end_index):
+        """Creating the mdList"""
+        for item in self.queryreturn[start_index:end_index]:
+            message_row = SwipeToDeleteItem(
+                text=item[0],
+            )
+            listItem = message_row.ids.content
+            listItem.secondary_text = item[1]
+            listItem.theme_text_color = "Custom"
+            listItem.text_color = ThemeClsColor
+            image = state.imageDir + "/text_images/{}.png".format(
+                avatarImageFirstLetter(item[0].strip()))
+            message_row.ids.avater_img.source = image
+            listItem.bind(on_release=partial(
+                self.addBook_detail, item[1], item[0], message_row))
+            message_row.ids.delete_msg.bind(on_press=partial(self.delete_address, item[1]))
+            self.ids.ml.add_widget(message_row)
+
+    def check_scroll_y(self, instance, somethingelse):
+        """Load data on scroll"""
+        if self.ids.scroll_y.scroll_y <= -0.0 and self.has_refreshed:
+            self.ids.scroll_y.scroll_y = 0.06
+            exist_addresses = len(self.ids.ml.children)
+            if exist_addresses != len(self.queryreturn):
+                self.update_addressBook_on_scroll(exist_addresses)
+            self.has_refreshed = (
+                True if exist_addresses != len(self.queryreturn) else False
+            )
+
+    def update_addressBook_on_scroll(self, exist_addresses):
+        """Load more data on scroll down"""
+        self.set_mdList(exist_addresses, exist_addresses + 5)
+
+    @staticmethod
+    def refreshs(*args):
+        """Refresh the Widget"""
+
+    # @staticmethod
+    def addBook_detail(self, address, label, instance, *args):
+        """Addressbook details"""
+        if instance.state == 'closed':
+            instance.ids.delete_msg.disabled = True
+            if instance.open_progress == 0.0:
+                obj = AddbookDetailPopup()
+                self.address_label = obj.address_label = label
+                self.address = obj.address = address
+                width = .9 if platform == 'android' else .8
+                self.addbook_popup = self.address_detail_popup(
+                    self.send_message_to, self.update_addbook_label, self.close_pop,
+                    width=width, obj=obj)
+                self.addbook_popup.auto_dismiss = False
+                self.addbook_popup.open()
+        else:
+            instance.ids.delete_msg.disabled = False
+
+    def delete_address(self, address, instance, *args):
+        """Delete inbox mail from inbox listing"""
+        self.ids.ml.remove_widget(instance.parent.parent)
+        # if len(self.ids.ml.children) == 0:
+        if self.ids.ml.children is not None:
+            self.ids.tag_label.text = ''
+        sqlExecute(
+            "DELETE FROM  addressbook WHERE address = '{}';".format(address))
```

Comment:
for all SQL statements:
- no need to use `;`
- don't use `{}` + format, pass arguments as an array and use `?`